regression: testdir fixture not found with pytester plugin loaded

After #9208, this:

```python
pytest_plugins = 'pytester'

def test_testdir(testdir):
    pass
```

fails with:

```
============================= test session starts ==============================
platform linux -- Python 3.9.7, pytest-7.0.0.dev294+g86446edc8, pluggy-1.0.0
rootdir: /home/florian/tmp/td
collected 1 item                                                               

test_td.py E                                                             [100%]

==================================== ERRORS ====================================
________________________ ERROR at setup of test_testdir ________________________
file /home/florian/tmp/td/test_td.py, line 3
  def test_testdir(testdir):
E       fixture 'testdir' not found
>       available fixtures: LineMatcher, _config_for_test, _pytest, _sys_snapshot, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, linecomp, monkeypatch, pytestconfig, pytester, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/florian/tmp/td/test_td.py:3
=========================== short test summary info ============================
ERROR test_td.py::test_testdir
=============================== 1 error in 0.01s ===============================
```

I bisected this to e6eac28f0eefd6eca641111f14a565f9ea15da1b.

Considering this as a showstopper for 7.0 given that this is [a very common](https://github.com/search?q=pytest_plugins+pytester&type=code) way of accessing the `testdir` fixture.

cc @bluetech

--- a/src/_pytest/legacypath.py
+++ b/src/_pytest/legacypath.py
@@ -400,12 +400,10 @@ def Node_fspath_set(self: Node, value: LEGACY_PATH) -> None:
     self.path = Path(value)
 
 
+@pytest.hookimpl
 def pytest_configure(config: pytest.Config) -> None:
     mp = pytest.MonkeyPatch()
     config.add_cleanup(mp.undo)
-
-    if config.pluginmanager.has_plugin("pytester"):
-        config.pluginmanager.register(LegacyTestdirPlugin, "legacypath-pytester")
 
     if config.pluginmanager.has_plugin("tmpdir"):
         # Create TmpdirFactory and attach it to the config object.
@@ -452,3 +450,14 @@ def pytest_configure(config: pytest.Config) -> None:
 
     # Add Node.fspath property.
     mp.setattr(Node, "fspath", property(Node_fspath, Node_fspath_set), raising=False)
+
+
+@pytest.hookimpl
+def pytest_plugin_registered(
+    plugin: object, manager: pytest.PytestPluginManager
+) -> None:
+    # pytester is not loaded by default and is commonly loaded from a conftest,
+    # so checking for it in `pytest_configure` is not enough.
+    is_pytester = plugin is manager.get_plugin("pytester")
+    if is_pytester and not manager.is_registered(LegacyTestdirPlugin):
+        manager.register(LegacyTestdirPlugin, "legacypath-pytester")

